packages/server/parseBody.ts

```diff
@@ -0,0 +1,22 @@
+import {HttpResponse} from 'uWebSockets.js'
+
+const defaultParser = (buffer: Buffer) => JSON.parse(buffer.toString())
+
+const parseBody = (res: HttpResponse, parser = defaultParser) => {
+  return new Promise<JSON | null>((resolve) => {
+    let buffer: Buffer
+    res.onData((ab, isLast) => {
+      const curBuf = Buffer.from(ab)
+      buffer = buffer ? Buffer.concat([buffer, curBuf]) : isLast ? curBuf : Buffer.concat([curBuf])
```

Comment:
this is a big gotcha with uWS.
each chunk (`ab`) reuses the previous chunk's buffer. so, we have to clone that chunk before we can get the next.
since we resolve with JSON (or a string), we don't have to clone the last buffer.
since we usually have single-chunk payloads, this is pretty efficient.
since we finish by writing a string from the entire buffer, we can save